app/graphql/resolvers/product_resolver.py

```diff
@@ -0,0 +1,36 @@
+import strawberry
+from typing import List, Optional, Any
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from app.models import ProductCategory
+from app.graphql.types.product_type import (
+    ProductType,
+    ProductCategoryType,
+    ProductSearchInput,
+)
+from strawberry.types import Info
+from app.services.search_service import search_products_service
+
+
+@strawberry.type
+class ProductResolver:
+    @strawberry.field
+    async def search_products(
+        self, info: Info[Any, Any], filters: Optional[ProductSearchInput] = None
```

Comment:
you can use `strawberry.Info` directly (and don't need to pass the arguments, they are optional)

```suggestion
        self, info: strawberry.Info, filters: ProductSearchInput | None = None
```

but if you create a context type, you can also do 

`strawberry.info[ContextType]` and then `info.context` will be typed 😊

Reply:
Thanks for the comment, I learned something new 😅